### ISSUE TITLE
Extending the hooks to a custom class

### ISSUE THREAD

**Issue by ajshortt** [directus/directus#1452]
## Version Info
Directus version and branch: 6.3.9
PHP version: 5.6.25
MySQL version: 14.14
Web server: Apache
OS name and version: macOS 10.12.4

## Expected Behavior
I want to perform tasks after a `table.insert` has fired.
Instead of having these within `configuration.php`, I want 
to include and instance an object to perform my tasks like so:
```
'table.insert' => function ($tableName, $record) {
    $obj = new ClassOfSomeSort($tableName, $record);
    $obj->publicMethod();   
},
```
## Actual Behavior
I'm getting errors with this  and when I try to include the class file

How can I solve this?

**Comment by WellingGuzman** [comment]
Hey @ajshortt, what kind of errors are you getting?

You can use classes, here is a [example](https://github.com/directus/docs/blob/3c78439710ddff69215d5cecbd09538954ba0b82/developer-guide/hooks.md#using-a-class-implementing-__invoke)

**Comment by benhaynes** [comment]
Hey @ajshortt – let us know if the example Welling provided fixes your issue. It's been a few months with no response, and we've released a new version (`6.4`), so I'll close this since the example code provided works.
